src/connectors/hierarchical-menu/connectHierarchicalMenu.js

```diff
@@ -113,93 +113,45 @@ export default function connectHierarchicalMenu(renderFn, unmountFn = noop) {
         this._refine = function(facetValue) {
           helper.toggleRefinement(hierarchicalFacetName, facetValue).search();
         };
 
-        // Bind createURL to this specific attribute
-        function _createURL(facetValue) {
-          return createURL(
-            helper.state.toggleRefinement(hierarchicalFacetName, facetValue)
-          );
-        }
-
         renderFn(
           {
-            items: [],
-            createURL: _createURL,
-            refine: this._refine,
+            ...this.getWidgetRenderState(renderState, initOptions)
+              .hierarchicalMenu[hierarchicalFacetName],
             instantSearchInstance,
-            widgetParams,
-            isShowingMore: false,
-            toggleShowMore: this.cachedToggleShowMore,
-            canToggleShowMore: false,
           },
           true
         );
       },
 
-      _prepareFacetValues(facetValues, state) {
+      _prepareFacetValues(facetValues) {
         return facetValues
           .slice(0, this.getLimit())
           .map(({ name: label, path: value, ...subValue }) => {
             if (Array.isArray(subValue.data)) {
-              subValue.data = this._prepareFacetValues(subValue.data, state);
+              subValue.data = this._prepareFacetValues(subValue.data);
             }
             return { ...subValue, label, value };
           });
       },
 
       render(renderOptions) {
-        const {
-          results,
-          state,
-          createURL,
-          instantSearchInstance,
-        } = renderOptions;
-
-        const facetValues =
-          results.getFacetValues(hierarchicalFacetName, { sortBy }).data || [];
-        const items = transformItems(
-          this._prepareFacetValues(facetValues),
-          state
-        );
-
-        // Bind createURL to this specific attribute
-        function _createURL(facetValue) {
-          return createURL(
-            state.toggleRefinement(hierarchicalFacetName, facetValue)
-          );
-        }
-
-        const maxValuesPerFacetConfig = state.maxValuesPerFacet;
-        const currentLimit = this.getLimit();
-        // If the limit is the max number of facet retrieved it is impossible to know
-        // if the facets are exhaustive. The only moment we are sure it is exhaustive
-        // is when it is strictly under the number requested unless we know that another
-        // widget has requested more values (maxValuesPerFacet > getLimit()).
-        // Because this is used for making the search of facets unable or not, it is important
-        // to be conservative here.
-        const hasExhaustiveItems =
-          maxValuesPerFacetConfig > currentLimit
-            ? facetValues.length <= currentLimit
-            : facetValues.length < currentLimit;
+        const { renderState, instantSearchInstance } = renderOptions;
 
         this.toggleShowMore = this.createToggleShowMore(renderOptions);
```

Comment:
what is this function used for? It seems to be using `cachedToggleShowMore` in the other occurrences

Reply:
That's a proxy to `cachedToggleShowMore` that keeps the same function reference across renders. It was introduced a long time ago.

Reply:
I see now, thanks!